Tasty now uses "." for namespaces instead of "/"

sevanspowell opened the pull request:
- Tasty now uses "." for namespaces instead of "/".
- Update the golden test appropriately.
- Add new location information for test failures.

Fixes issue #2.

sevanspowell committed:
Tasty now uses "." for namespaces instead of "/"

- Tasty now uses "." for namespaces instead of "/", updated the golden test
  appropriately.
- Added min version bound for tasty. According to
  https://github.com/feuerbach/tasty/blob/master/core/CHANGELOG.md the format of
  the TAP output changed in version 1.1, so I've added a min version bound to
  ensure that we only build against tasty versions with the new output format.
- Updated the stackage resolver so that we get a version of tasty >= 1.1 (which
  is the minimum required version for the new test format).

MichaelXavier commented:
@sevanspowell would it be a good idea to set a min bound in the test suite for tasty since the golden file's expectations is linked to the tasty version where this behavior changed?

sevanspowell commented:
@MichaelXavier Yeah, as soon as all those builds failed I realized I wasn't considering everything. I haven't had a chance to fix it yet, but will do soon. Thanks.

sevanspowell commented:
Hey @MichaelXavier, I've placed a higher min bound on tasty and bumped the stack resolver to the oldest LTS version that satisfies the new tasty version bound.

The "ARGS=''" build passes but any LTS older than 12.0 is going to fail, what can I do about this?

Commit:
Update Stackage resolvers used in TravisCI

- The CI system currently builds with LTS versions 2, 3, 4, 5, 6, nightly and
  default. Kept nightly and default, but changed to use later LTS versions 12,
  13 and 14 instead. This is because the new tasty min bound (>= 1.1) can only
  be satisfied on resolvers >= 12.

sevanspowell commented:
For now I've updated the Stackage resolvers Travis uses.

MichaelXavier commented:
@sevanspowell I'm okay with just not building on LTS < 12. Also seems like we should have the default LTS in stack.yaml for the package be the latest version of the latest LTS (14?). Once you do that, I think this should be ready to merge. Thanks!

Commit:
Move stack resolver to lts-14.12

- Moved the stack resolver to the latest lts resolver at the time of writing.

sevanspowell commented:
Great!

I've moved the stack resolver to the latest LTS version at the time of writing (14.12).

Thanks for your help with this.

MichaelXavier commented:
This is released as tasty-tap-0.1.0. Thanks!

sevanspowell commented:
Thanks!
